main.go

```diff
@@ -0,0 +1,55 @@
+// grep | (c) 2020 NETWAYS GmbH | GPLv2+
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+)
+
+func main() {
+	patternParameter := flag.String("e", "", "use PATTERN for matching.")
+	flag.Parse()
+
+	if *patternParameter == "" {
+		fmt.Fprintln(os.Stderr, "grep: at least the -e parameter is needed.")
+		os.Exit(1)
+	}
+
+	re, regErr := regexp.Compile(*patternParameter)
+	if regErr != nil {
+		fmt.Fprintln(os.Stderr, regErr)
+		os.Exit(1)
+	}
+
+	buf := bufio.NewReader(os.Stdin)
+
+	for {
+		data, dataErr := buf.ReadBytes('\n')
+		if dataErr != nil && dataErr != io.EOF {
+			fmt.Fprintln(os.Stderr, dataErr)
+			os.Exit(1)
+		}
+
+		if len(data) == 0 {
+			break
+		}
+
+		if !bytes.HasSuffix(data, []byte{'\n'}) {
+			data = append(data, '\n')
+		}
```

Comment:
Do you always need this block? Or just on certain conditions?

Reply:
i need to check every single line so i think I need it always

Reply:
The block eventually modifies `data`. Do you always need `data` modified this way? Or only if... ?

Reply:
data is a line of a text file and if the line has no newline at the end I add one and if not then not. I dont get your point

Reply:
**What exactly** do you need the newline for?

Reply:
for my output to be equal to greps output

Reply:
Right! Your **output**! But you don't output every line, do you?

Reply:
I output the line where my regex expression is found in? do you want me to put this code into my if re.match?

Reply:
Exactly.